lake/journal/store.go

```diff
@@ -74,34 +76,41 @@ func (s *Store) load(ctx context.Context) error {
 		return err
 	}
 	s.mu.RLock()
-	at := s.at
+	current := s.at
 	s.mu.RUnlock()
-	if head == at {
+	if head == current {
 		return nil
 	}
-	r, err := s.journal.OpenAsZNG(ctx, zed.NewContext(), head, 0)
+	at, table, err := s.getSnapshot(ctx)
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return err
+	}
+	r, err := s.journal.OpenAsZNG(ctx, zed.NewContext(), head, at)
 	if err != nil {
 		return err
 	}
 	defer r.Close()
-	table := make(map[string]Entry)
 	for {
-		rec, err := r.Read()
+		v, err := r.Read()
 		if err != nil {
 			return err
 		}
-		if rec == nil {
+		if v == nil {
 			now := time.Now()
 			s.mu.Lock()
 			s.table = table
 			s.at = head
 			s.loadTime = now
 			s.mu.Unlock()
+			// Reduce the amount of times we write snapshots to disk by only writing when there are 10 or more new entries since
+			// the last snapshot.
+			if head-at > 10 {
```

Comment:
Comment says "10 or more" but code says `head-at > 10`, and that means more than 10, right?

Reply:
Yes, will update comment.